src/macaron/vsa/vsa.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""VSA schema and generation."""
+
+from __future__ import annotations
+
+import base64
+import json
+from datetime import datetime
+from enum import StrEnum
+from importlib import metadata as importlib_metadata
+from typing import Any, TypedDict
+
+
+class Vsa(TypedDict):
+    """The Macaron Verification Summary Attestation.
+
+    For reference, see:
+    * `SLSA <https://slsa.dev/spec/v1.0/verification_summary>`_.
+    * `in-toto <https://github.com/in-toto/attestation/blob/main/spec/predicates/vsa.md>`_.
+    """
+
+    #: The payload type. Following in-toto, this is always ``application/vnd.in-toto+json``.
+    payloadType: str  # noqa: N815
+
+    #: The payload of the VSA, base64 encoded.
+    payload: str
+
+
+class VsaStatement(TypedDict):
+    """The Statement layer of a Macaron VSA.
+
+    For reference, see:
+    * in-toto Statement layer specification: https://github.com/in-toto/attestation/blob/main/spec/v1/statement.md.
+    """
+
+    _type: str
+    subject: list[dict]
+    predicateType: str  # noqa: N815
+    predicate: VsaPredicate
```

Comment:
The documentation for these attributes is missing.

Reply:
See bcfea7f5aea0124c0421336ff1c85ac131deb0b6.